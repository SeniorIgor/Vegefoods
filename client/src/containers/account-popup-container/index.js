import React from 'react';
import { compose } from 'redux';
import { connect } from 'react-redux';

import AccountPopup from '../../components/account-popup';
import { withAuth } from '../../components/hoc';

export const AccountPopupContainer = ({ logout, isOpenAuthPopup, closeMenu }) => {

	return (
		<AccountPopup isOpen={isOpenAuthPopup} logout={logout} closeMenu={closeMenu} />
	);
}

export const mapStateToProps = ({ accountPopup: { isOpenAuthPopup } }, { logout }) =>
	({ logout, isOpenAuthPopup: isOpenAuthPopup });

export default compose(
	withAuth,
	connect(mapStateToProps),
)(AccountPopupContainer);