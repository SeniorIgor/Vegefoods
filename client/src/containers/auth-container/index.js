import React, { Component, Fragment } from 'react';
import { compose, bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import AuthPopup from '../../components/auth-popup/auth-popup';
import { AuthProvider } from '../../components/auth-context';
import { AuthService } from '../../services';
import { withProductService } from '../../components/hoc';
import {
	fetchCart, clearCart, fetchFavorites, clearFavorites,
	fetchProfile, clearProfile, updateUser, authLogout,
	viewAuthPopup, closeAuthPopup
} from '../../actions';

export class AuthContainer extends Component {

	componentDidMount() {
		this.props.setAuthHandler(this.handlerAuthErr);

		const user = AuthService.getUser();

		if (user) {
			this.props.updateUser(user);
			this.updateData(user.token);
		}
	}

	updateData = (token) => {
		const { fetchCart, fetchFavorites, fetchProfile } = this.props;

		fetchCart(token);
		fetchFavorites(token);
		fetchProfile(token);
	}

	handlerAuthErr = () => {
		this.logout();
		this.onViewPopup();
	}

	onViewPopup = () => {
		document.body.classList.add('lock');
		this.props.viewAuthPopup();
	}

	onClosePopup = () => {
		document.body.classList.remove('lock');
		this.props.closeAuthPopup();
	}

	login = (user) => {
		AuthService.logIn(user);
		this.onClosePopup();
		this.updateData(user.token);
	}

	logout = () => {
		const { authLogout, clearCart, clearFavorites, clearProfile } = this.props;

		AuthService.logOut();
		authLogout();
		clearCart();
		clearFavorites();
		clearProfile();
	}

	render() {
		const { isView, isAuth, token } = this.props;

		const value = {
			isAuth,
			token,
			login: this.login,
			logout: this.logout,
			onViewPopup: this.onViewPopup,
			onClosePopup: this.onClosePopup,
			handlerAuthErr: this.handlerAuthErr,
		}

		return (
			<AuthProvider value={value}>
				<Fragment>
					{this.props.children}
					<AuthPopup isView={isView} onClose={this.onClosePopup} />
				</Fragment>
			</AuthProvider>
		);
	}
}

export const mapMethodsToProps = ({ setAuthHandler, getCart, getFavorites, getProfile }) => ({
	setAuthHandler, getCart, getFavorites, getProfile
});

export const mapStateToProps = ({ login: { isAuth, token }, authPopup: { isView } }) => ({ isAuth, isView, token });

export const mapDispatchToProps = (dispatch,
	{ getCart, getFavorites, getProfile }) => bindActionCreators({
		fetchCart: fetchCart(getCart),
		fetchFavorites: fetchFavorites(getFavorites),
		fetchProfile: fetchProfile(getProfile),
		clearProfile, clearCart, clearFavorites, viewAuthPopup,
		closeAuthPopup, updateUser, authLogout
	}, dispatch);


export default compose(
	withProductService(mapMethodsToProps),
	connect(mapStateToProps, mapDispatchToProps),
)(AuthContainer);