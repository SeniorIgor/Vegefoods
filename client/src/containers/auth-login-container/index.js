import React, { Component } from 'react';
import { bindActionCreators, compose } from 'redux';
import { connect } from 'react-redux';

import SpinnerWrapper from '../../components/spinner-wrapper';
import ErrorIndicator from '../../components/error-indicator';
import { withProductService, withAuth } from '../../components/hoc';
import {
	fetchLogin, loadedToStorage, updateLogin,
	allowLogin, clearLogin
} from '../../actions';
import { AuthLogin } from '../../components/auth-forms';

export class AuthLoginContainer extends Component {

	componentDidUpdate(prevProps) {
		const { isNewAuth, userId, token, allowCreate, email, password } = this.props.loginData;

		if (isNewAuth && isNewAuth !== prevProps.loginData.isNewAuth) {
			this.props.login({ userId, token });
			this.props.loadedToStorage();
		}

		if (allowCreate && prevProps.loginData.allowCreate !== allowCreate) {
			this.props.loginUser({ email, password });
		}
	}

	componentWillUnmount() {
		this.props.clearLogin();
	}

	render() {
		const { loginData, updateValue, allowLogin } = this.props;
		const { loading, error, email, password, errors } = loginData;

		const validateError = (error && error.name === 'ValidationError') ? error : null;
		if (error && !validateError) return <ErrorIndicator />;

		return (
			<SpinnerWrapper loading={loading}>
				<AuthLogin loginUser={allowLogin} err={validateError} {...{ email, password, errors }}
					updateValue={updateValue} />
			</SpinnerWrapper>
		)
	}
}

export const mapMethodsToProps = ({ loginUser }) => ({ loginUser });

export const mapStateToProps = ({ login: loginData }) => ({ loginData });

export const mapDispatchToProps = (dispatch, { loginUser }) => {
	return bindActionCreators({
		loginUser: fetchLogin(loginUser),
		loadedToStorage, updateValue: updateLogin, allowLogin, clearLogin
	}, dispatch);
}

export default compose(
	withProductService(mapMethodsToProps),
	withAuth,
	connect(mapStateToProps, mapDispatchToProps),
)(AuthLoginContainer);