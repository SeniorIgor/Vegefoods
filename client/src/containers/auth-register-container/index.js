import React, { Component } from 'react';
import { compose, bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import SpinnerWrapper from '../../components/spinner-wrapper';
import ErrorIndicator from '../../components/error-indicator';
import { withProductService } from '../../components/hoc';
import {
	fetchRegister, updateRegister, allowRegister,
	clearRegister
} from '../../actions';
import { AuthRegister } from '../../components/auth-forms';

export class AuthRegisterContainer extends Component {

	componentDidUpdate(prevProps) {
		const { name, email, password, confirm, allowCreate } = this.props.register;

		if (allowCreate && prevProps.register.allowCreate !== allowCreate) {
			this.props.registerUser({ name, email, password, confirm });
		}
	}

	componentWillUnmount() {
		this.props.clearRegister();
	}

	render() {
		const { register, updateValue, allowRegister } = this.props;
		const { loading, error, ...fields } = register;

		const validateError = (error && error.name === 'ValidationError') ? error : null;
		if (error && !validateError) return <ErrorIndicator />;

		return (
			<SpinnerWrapper loading={loading}>
				<AuthRegister {...fields} updateValue={updateValue}
					errorData={validateError} registerUser={allowRegister} />
			</SpinnerWrapper>
		)
	}
}

export const mapMethodsToProps = ({ registerUser }) => ({ registerUser });

export const mapStateToProps = ({ register }) => ({ register });

export const mapDispatchToProps = (dispatch, { registerUser }) => {
	return bindActionCreators({
		registerUser: fetchRegister(registerUser),
		updateValue: updateRegister,
		allowRegister, clearRegister,
	}, dispatch);
}

export default compose(
	withProductService(mapMethodsToProps),
	connect(mapStateToProps, mapDispatchToProps),
)(AuthRegisterContainer);