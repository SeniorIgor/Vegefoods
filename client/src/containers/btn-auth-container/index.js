import React from 'react';
import { compose } from 'redux';
import { connect } from 'react-redux';

import BtnAuth from '../../components/btn-auth';
import { withAuth } from '../../components/hoc';
import * as actions from '../../actions/account-popup';

export const BtnAuthContainer = ({ isAuth, onViewPopup, viewAccountPopup, closeAccountPopup }) => {
	return (
		<BtnAuth isAuth={isAuth} onViewAuth={onViewPopup} onViewAccount={viewAccountPopup}
			onCloseAccount={closeAccountPopup} />
	);
}

export default compose(
	withAuth,
	connect(null, actions),
)(BtnAuthContainer);