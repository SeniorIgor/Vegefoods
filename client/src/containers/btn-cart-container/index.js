import React from 'react';
import { connect } from 'react-redux';

import BtnCart from '../../components/btn-cart';

export const BtnCartContainer = ({ countItems }) => {

	return (
		<BtnCart countItems={countItems} />
	);
}

export const mapStateToProps = ({ shoppingCart: { cartItems } }) => ({
	countItems: cartItems.length,
})

export default connect(mapStateToProps)(BtnCartContainer);