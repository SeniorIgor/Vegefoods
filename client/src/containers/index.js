import ProductListContainer from './product-list-container';
import CategoryListContainer from './category-list-container';
import PaginationContainer from './pagination-container';
import AuthContainer from './auth-container';
import AuthRegisterContainer from './auth-register-container';
import AuthLoginContainer from './auth-login-container';
import AccountPopupContainer from './account-popup-container';
import CartListContainer from './cart-list-container';
import BtnQuantityContainer from './btn-quantity-container';
import BtnFavoritesContainer from './btn-favorites-container';
import FavoritesListContainer from './favorites-list-container';
import BtnAuthContainer from './btn-auth-container';
import BtnCartContainer from './btn-cart-container';
import BtnNavFavoritesContainer from './btn-nav-favorites-container';
import BtnMainMenuContainer from './btn-main-menu-container';
import MainMenuContainer from './main-menu-container';
import MainMenuCloseContainer from './main-menu-close-container';
import FieldPhoneContainer from './field-phone-container';
import OrderingContainer from './ordering-container';
import OrderDetailsContainer from './order-details-container';
import BtnCartRemoveContainer from './btn-cart-remove-container';
import ContactsContainer from './contacts-container';
import OrderListContainer from './order-list-container';
import OrderItemContainer from './order-item-container';
import OrderingViewContainer from './ordering-view-container';

export {
	ProductListContainer,
	CategoryListContainer,
	PaginationContainer,
	AuthContainer,
	AuthRegisterContainer,
	AuthLoginContainer,
	AccountPopupContainer,
	CartListContainer,
	BtnQuantityContainer,
	BtnFavoritesContainer,
	FavoritesListContainer,
	BtnAuthContainer,
	BtnCartContainer,
	BtnNavFavoritesContainer,
	BtnMainMenuContainer,
	MainMenuContainer,
	MainMenuCloseContainer,
	FieldPhoneContainer,
	OrderingContainer,
	OrderDetailsContainer,
	BtnCartRemoveContainer,
	ContactsContainer,
	OrderListContainer,
	OrderItemContainer,
	OrderingViewContainer,
};