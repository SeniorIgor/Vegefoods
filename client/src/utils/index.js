import { useRoutes } from './routes';
import { ValidationError, AuthorizationError } from './errors';
import {
	validEmailRegex,
	validateForm,
	fieldLengthError,
	emailError,
	confirmError,
} from './validation';
import {
	formatPrice,
	discountPrice,
	formatNumber,
	getWeekDay,
	getEarlyDate,
	getDeliveryDates,
	getDeliveryTime,
	formatDate,
	formatDateTime,
	formatPhoneNumber,
} from './format-data';
import { getProductsForPage } from './product-list';
import { setUp } from './test-utils';


export {
	// format-data
	formatNumber,
	discountPrice,
	formatPrice,
	formatPhoneNumber,
	formatDate,
	formatDateTime,
	// routes
	useRoutes,
	// validation
	validEmailRegex,
	validateForm,
	fieldLengthError,
	emailError,
	confirmError,
	// date
	getWeekDay,
	getEarlyDate,
	getDeliveryDates,
	getDeliveryTime,
	// errors
	ValidationError,
	AuthorizationError,
	// product-list
	getProductsForPage,
	// test-utils
	setUp,
}