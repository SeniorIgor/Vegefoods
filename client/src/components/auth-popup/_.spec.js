import React from 'react';
import { shallow } from 'enzyme';
import { setUp } from '../../utils';

import { render as renderComponent } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import AuthPopup from './index';
import { AuthLoginContainer, AuthRegisterContainer } from '../../containers';

const initialState = {
	isView: false,
	onClose: () => { },
}

const render = setUp(AuthPopup, shallow, initialState);

describe('AUTH_POPUP: ', () => {

	let component;
	// let instance;
	beforeEach(() => {
		component = render();
		// instance = component.instance();
	})

	it('render component (isView=false)', () => {
		expect(component.find(".auth-overlay")).toHaveLength(1);
		expect(component.find(".auth-overlay_active")).toHaveLength(0);
	})

	it('render component (isView=false)', () => {
		const newComponent = render({ isView: true });
		expect(newComponent.find(".auth-overlay_active")).toHaveLength(1);
	})

	describe('HANDLERS: ', () => {

		it('should select Login form', () => {
			const btns = component.find(".auth-overlay__toggle-btn");
			btns.at(0).simulate("click");
			expect(component.find(AuthLoginContainer)).toHaveLength(1);
			expect(component.find(AuthRegisterContainer)).toHaveLength(0);

			btns.at(1).simulate("click");
			expect(component.find(AuthLoginContainer)).toHaveLength(0);
			expect(component.find(AuthRegisterContainer)).toHaveLength(1);
		})

		it('should close popup (click on btn)', () => {
			const onClose = jest.fn();
			const newComponent = render({ onClose });
			newComponent.find(".auth-overlay__close-btn").simulate("click");
			expect(onClose).toHaveBeenCalledTimes(1);
		})

		it('should close popup (click on area)', () => {
			const onClose = jest.fn();
			const newComponent = render({ onClose });
			newComponent.find(".auth-overlay").simulate("click", {
				target: 1,
				currentTarget: 1,
			});
			expect(onClose).toHaveBeenCalledTimes(1);
		})

		it('should not close popup (click inside area)', () => {
			const onClose = jest.fn();
			const newComponent = render({ onClose });
			newComponent.find(".auth-overlay").simulate("click", {
				target: 1,
				currentTarget: 2,
			});
			expect(onClose).toHaveBeenCalledTimes(0);
		})

	})


})