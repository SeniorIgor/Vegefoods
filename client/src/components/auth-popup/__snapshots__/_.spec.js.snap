// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AUTH_POPUP:  should render component 1`] = `
<div
  className="auth-overlay"
  onClick={[Function]}
>
  <div
    className="auth-overlay__wrapper"
  >
    <div
      className="auth-overlay__popup"
    >
      <button
        className="auth-overlay__close-btn"
        onClick={[Function]}
      />
      <div
        className="auth-overlay__toggle-wrap"
      >
        <div
          className="auth-overlay__toggle-btn auth-overlay__toggle-btn_active"
          onClick={[Function]}
        >
          Вход
        </div>
        <div
          className="auth-overlay__toggle-btn"
          onClick={[Function]}
        >
          Регистрация
        </div>
      </div>
      <Component />
    </div>
  </div>
</div>
`;
